kube/service_test.go

```diff
@@ -143,3 +245,258 @@ func TestHeadlessServiceOK(t *testing.T) {
+		return
+	}
+
+	portDef := role.Run.ExposedPorts[0]
+	if !assert.NotNil(portDef) {
+		return
+	}
+	service, err := NewClusterIPService(role, true, false, ExportSettings{
+		CreateHelmChart: true,
+	})
+	if !assert.NoError(err) {
+		return
+	}
+	require.NotNil(t, service)
+	assert.Equal("{{ if .Values.services.loadbalanced }} LoadBalancer {{ else }} ClusterIP {{ end }}", service.Get("spec", "type").String())
+	assert.Equal("None", service.Get("spec", "clusterIP").String())
+
+	serviceYAML, err := testhelpers.RenderNode(service, nil)
+	if !assert.NoError(err) {
+		return
+	}
+
+	expected := `---
+apiVersion: "v1"
+kind: "Service"
+metadata:
+  name: "myrole-set"
+spec:
+  clusterIP: "None"
+  ports:
+  - name: "http"
+    port: 80
+    protocol: "TCP"
+    targetPort: 0
+  - name: "https"
+    port: 443
+    protocol: "TCP"
+    targetPort: 0
+  selector:
+    skiff-role-name: "myrole"
+  type:  ClusterIP 
+`
+	assert.Equal(expected, string(serviceYAML))
+}
+
+func TestHeadlessServiceHelmOKConfigured(t *testing.T) {
+	assert := assert.New(t)
+
+	manifest, role := serviceTestLoadRole(assert, "exposed-ports.yml")
+	if manifest == nil || role == nil {
+		return
+	}
+
+	portDef := role.Run.ExposedPorts[0]
+	if !assert.NotNil(portDef) {
+		return
+	}
+	service, err := NewClusterIPService(role, true, false, ExportSettings{
+		CreateHelmChart: true,
+	})
+	if !assert.NoError(err) {
+		return
+	}
+	require.NotNil(t, service)
+	assert.Equal("{{ if .Values.services.loadbalanced }} LoadBalancer {{ else }} ClusterIP {{ end }}", service.Get("spec", "type").String())
+	assert.Equal("None", service.Get("spec", "clusterIP").String())
+
+	config := map[string]interface{}{
+		"Values.services.loadbalanced": "true",
+	}
+
+	serviceYAML, err := testhelpers.RenderNode(service, config)
+	if !assert.NoError(err) {
+		return
+	}
+
+	expected := `---
+apiVersion: "v1"
+kind: "Service"
+metadata:
+  name: "myrole-set"
+spec:
+  ports:
+  - name: "http"
+    port: 80
+    protocol: "TCP"
+    targetPort: 0
+  - name: "https"
+    port: 443
+    protocol: "TCP"
+    targetPort: 0
+  selector:
+    skiff-role-name: "myrole"
+  type:  LoadBalancer 
+`
+	assert.Equal(expected, string(serviceYAML))
+}
+
+func TestPublicServiceOK(t *testing.T) {
+	assert := assert.New(t)
+
+	manifest, role := serviceTestLoadRole(assert, "exposed-ports.yml")
+	if manifest == nil || role == nil {
+		return
+	}
+
+	portDef := role.Run.ExposedPorts[0]
+	if !assert.NotNil(portDef) {
+		return
+	}
+	service, err := NewClusterIPService(role, false, true, ExportSettings{})
+	if !assert.NoError(err) {
+		return
+	}
+	require.NotNil(t, service)
+	assert.Equal("ClusterIP", service.Get("spec", "type").String())
+	assert.Nil(service.Get("spec", "clusterIP"))
+
+	yamlConfig := &bytes.Buffer{}
+	if err := helm.NewEncoder(yamlConfig).Encode(service); !assert.NoError(err) {
+		return
+	}
+	var expected, actual interface{}
+	if !assert.NoError(yaml.Unmarshal(yamlConfig.Bytes(), &actual)) {
```

Comment:
Isn't that `testhelpers.RoundtripNode`?

Reply:
No. That test has no RenderNode which would be needed for a full Roundtrip.
That test is actually one I did not touch, i.e. existed before, in the same manner.
I might check if it can be rewritten for Roundtrip and if that is simpler.
However all the work to address the other comments will be more important.

Reply:
Reverting my statement. It is indeed Roundtrip with a config of `nil`.

Reply:
Done with commit abbd63a3a9fbf53076745a0609423858731db857